libcudacxx/include/cuda/barrier

```diff
@@ -60,14 +61,10 @@ void cp_async_bulk_global_to_shared(void *__dest, const void *__src, _CUDA_VSTD:
     _LIBCUDACXX_DEBUG_ASSERT(__isShared(__dest), "Destination must be shared memory address.");
     _LIBCUDACXX_DEBUG_ASSERT(__isGlobal(__src),  "Source must be global memory address.");
 
-    asm volatile(
-        "cp.async.bulk.shared::cluster.global.mbarrier::complete_tx::bytes [%0], [%1], %2, [%3];\n"
-        :
-        : "r"(static_cast<_CUDA_VSTD::uint32_t>(__cvta_generic_to_shared(__dest))),
-          "l"(static_cast<_CUDA_VSTD::uint64_t>(__cvta_generic_to_global(__src))),
-          "r"(__size),
-          "r"(static_cast<_CUDA_VSTD::uint32_t>(__cvta_generic_to_shared(::cuda::device::barrier_native_handle(__bar))))
-        : "memory");
+    _CUDA_VPTX::cp_async_bulk(
```

Comment:
I believe this should be guarded by the PTX ISA level too

Also we should add that restriction to the comment above

Reply:
Yes, I agree. Those guards are added in #1359 right? So if I add them here, we create a conflict with #1359, and if we don't then they will be added when #1359 is merged.